go/binlog/gomysql_reader.go

```diff
@@ -142,15 +153,46 @@ func (this *GoMySQLReader) StreamEvents(canStopStreaming func() bool, entriesCha
 			defer this.currentCoordinatesMutex.Unlock()
 			this.currentCoordinates.LogPos = int64(ev.Header.LogPos)
 		}()
-		if rotateEvent, ok := ev.Event.(*replication.RotateEvent); ok {
+
+		switch event := ev.Event.(type) {
+		case *replication.PreviousGTIDsEvent:
+			if !this.migrationContext.UseGTIDs {
+				continue
+			}
+			func() {
+				this.currentCoordinatesMutex.Lock()
+				defer this.currentCoordinatesMutex.Unlock()
+				if err := this.currentCoordinates.GTIDSet.Update(event.GTIDSets); err != nil {
+					this.migrationContext.Log.Errorf("Failed to parse GTID set: %v", err)
+				}
+			}()
+		case *replication.GTIDEvent:
+			if !this.migrationContext.UseGTIDs {
+				continue
+			}
+			sid, err := uuid.FromBytes(event.SID)
+			if err != nil {
+				return err
+			}
+			func() {
+				this.currentCoordinatesMutex.Lock()
+				defer this.currentCoordinatesMutex.Unlock()
+				this.nextCoordinates = this.currentCoordinates
+				interval := gomysql.Interval{Start: event.GNO, Stop: event.GNO + 1}
+				this.nextCoordinates.GTIDSet.AddSet(gomysql.NewUUIDSet(sid, interval))
```

Comment:
this assumes that each GTID event is always one and only one GTID.  Which sounds like it's probably correct, I just wanted to ask if that was something you specifically checked on when writing this code.